modules/data-streams/src/main/java/org/elasticsearch/datastreams/DataStreamFeatures.java

```diff
@@ -8,19 +8,28 @@
 
 package org.elasticsearch.datastreams;
 
+import org.elasticsearch.Version;
 import org.elasticsearch.action.admin.indices.rollover.LazyRolloverAction;
 import org.elasticsearch.action.datastreams.autosharding.DataStreamAutoShardingService;
 import org.elasticsearch.datastreams.lifecycle.health.DataStreamLifecycleHealthInfoPublisher;
 import org.elasticsearch.features.FeatureSpecification;
 import org.elasticsearch.features.NodeFeature;
 
+import java.util.Map;
 import java.util.Set;
 
 /**
  * Provides the features for data streams that this version of the code supports
  */
 public class DataStreamFeatures implements FeatureSpecification {
 
+    public static final NodeFeature SUPPORTS_DATA_STREAM_LIFECYCLE = new NodeFeature("supports_data_stream_lifecycle");
```

Comment:
Looking around, the string name for these kinds of node features follow a dotted and hierarchical structure. Maybe we should go with

```suggestion
    public static final NodeFeature DATA_STREAM_LIFECYCLE = new NodeFeature("data_stream.rollover.lifecycle");
```